src/states/node.rs

```diff
@@ -1442,7 +1442,13 @@ impl Node {
             return Ok(());  // Avoid swarming back out to our own group.
         }
 
-        let (new_sent_to, target_peer_ids) = try!(self.get_targets(routing_msg, route, sent_to));
+        let exclude = if routing_msg.dst.is_group() {
```

Comment:
I think we can exclude `hop` even then: They will have already handled it anyway.